### ISSUE TITLE
Moving tasks related to development back to GH

### ISSUE THREAD

**Issue by venkatd** [coderly/process#63]
What is the advantage of Asana currently for having code related tasks in there?

What we'd have to do:
- Link PRs to each user story in Asana
- Use GH milestones
- Potentially use a product like http://sweepboard.com/. I know the guy behind that so I can give him feedback directly if it doesn't suit our purposes yet.

**Comment by joshsmith** [comment]
The advantage of Asana, to my mind, is having a consistent place for _all_ tasks of any kind. I absolutely hate having to look at more than one place to know what needs done.

**Comment by idelahoz** [comment]
@venkatd so in the way you're saying we only keep user stories on asana?

**Comment by venkatd** [comment]
@JoshSmith could you comment on the JTBD that you have to handle when looking at Asana with our current process?

**Comment by venkatd** [comment]
@idelahoz yes that's what I was suggesting.

**Comment by venkatd** [comment]
I don't feel strongly one way or another. But I wanted to frame the discussion this way to get us thinking about what Asana is used for.

**Comment by joshsmith** [comment]
Asana is where I know what I'm doing at any given time, and what everyone else is doing. Asana is like my task-focused second brain. If you break that up, it's like having to context switch every time I try to answer the question: "what should I do next?"

**Comment by venkatd** [comment]
If we can figure out what JTBD Asana is satisfying, we can be very targeted with our process improvements.

Examples:
- What quick things can I do to move things forward on GH (quick questions, code reviews, etc)?
- Is anything blocked?
- Are we on track with user story X?

@JoshSmith "What task should I work on next?" is another example of a JTBD. If we can unpack "where I know what I'm doing at any given time, and what everyone else is doing" into the component parts, we will be able to improve the process and uncover pains--even in the case where it makes sense to stick with Asana.

In other words, if we know all the specific jobs we are trying to accomplish then it becomes a matter of identifying if any of those jobs are currently painful and improving the process to remove those pains.

**Comment by joshsmith** [comment]
Asana satisfies a lot of those at once. This feels very abstract to me and I tend to have trouble brainstorming these out of nowhere. It would kill me to take all these tasks out of Asana because I can't reliably answer the question "what should I do next?"

**Comment by joshsmith** [comment]
Besides, the RFC at hand is "Moving tasks related to development back to GH", and from that perspective this is a big :-1: from me.

**Comment by venkatd** [comment]
@JoshSmith I'm not suggesting we take the tasks out of Asana. Honestly I'm leaning toward keeping Asana.

Maybe this is a separate discussion, but I am more interested in honing in on what Asana is actually doing for us. If we pay attention to what job we are trying to accomplish when we fire up Asana, then I'd like to build a list of those jobs.

If this sounds abstract let me try to give an example.

Suppose I open up Asana to see what task I should work on next. In that moment, I can pay attention to what is going on in my head and think "I just opened up Asana to figure out what I should work on next. This is one job I try to get done."

Maybe an hour later "I just checked Asana to see if there were any tasks that need code reviews. Looks like I'm wanting to know what tasks are awaiting review right now"

So if we build up a list of these things then we can be more targeted with improvements. We can realize, "oh we have a lot of pain around knowing which tasks are awaiting review" and improve that process.

Not sure where to discuss something like this but either way I can start doing this exercise myself to get things going. I think there is a benefit to it but it's probably easier to show it in practice versus discussing it.

**Comment by joshsmith** [comment]
> I think there is a benefit to it but it's probably easier to show it in practice versus discussing it.

100% agreed. I just can't wrap my head around it when we're not discussing a specific instance.

**Comment by venkatd** [comment]
Let me close this since I think we're on the same page. Basically I'm asking us all to pay closer attention to the jobs we are trying to accomplish if that makes sense with the examples from above.
